RevenueCatUI/CustomerCenter/Actions/CustomerCenterView+Actions.swift

```diff
@@ -346,6 +361,12 @@ extension View {
     ) -> some View {
         return self.modifier(CustomerCenterView.OnPromotionalOfferSuccess(handler: handler))
     }
+
+    func onSubscriptionCancelled(
```

Comment:
I don't think adding this is what we should be doing because we won't trigger this when the subscription is cancelled, but when the button is pressed. We don't have a good way for detecting the subscription was actually cancelled. 

We are already triggering `onShowingManageSubscription` when the `Cancel subscription` button or the `Change plans` is selected, and we also have the `onButtonSelected` modifier as well. So I believe this is redundant informatino. 

I think we should instead just reload when that action happens, without adding any new modifier.

Reply:
Hmm I've been thinking... did cancelling trigger a new CustomerInfo in the CustomerInfo listener? If so, maybe we would want to listen to changes in that listener and just reload whenever a new version is received (just don't remember if cancellations show there immediately)

Reply:
Hmm cancellations by themselves won't trigger the listener, but reloading the customer info itself could, if the backend has had enough time to detect the cancellation.

Reply:
I've found https://developer.apple.com/documentation/storekit/transaction/updates

I guess that when the cancelled starts, we could listen to this for a couple of minutes?

Reply:
We could do something like:

```swift
    func listenToTransactionUpdates() async {
        for await update in StoreKit.Transaction.updates {
            await loadScreen()
        }
    }
```

1. We set it up once the customer center is loaded
2. If someone enters the cancel flow, we add a "waiting for cancellation" label (or something better) for a couple of seconds

Reply:
Although I am not sure this will work, I see there are a couple of options:
- https://developer.apple.com/documentation/storekit/transaction/latest(for:)
- https://developer.apple.com/documentation/storekit/transaction/all

Since we know the productIdentifier, we could use `latest(for:)` to see if there are any updates, and maybe subscribe to either `.all` or `.updates`

I am not completely sure it will work, but I can test it out and see what happens when cancelling

Reply:
In the end this didn't work. I've taken a complete different approach, relying on `syncPurchases` right after the management screen is closed